### DISCUSSION TITLE
prerequisites

### DISCUSSION THREAD

**Discussion by sugizo** [matt-42/lithium#52] [Q&A]
**steps in terminal**
```
curl -L -O -C - https://raw.githubusercontent.com/matt-42/lithium/master/cli/li 
cat << EOF > hello_world.cc
#include "lithium_http_server.hh"

int main() {

  // Define an api with one HTTP GET route.
  li::http_api api;
  api.get("/hello_world") = [&](li::http_request& request, li::http_response& response) {
    response.write("hello world.");
  };

  // Start a http server.
  li::http_serve(api, 12345);

}
li run 
EOF
chmod +x ./li
./li run ./hello_world.cc
```

**result**
```
Traceback (most recent call last):
  File "./li", line 201, in <module>
    args.func(args)
  File "./li", line 151, in run_command
    build_docker_image_if_missing()
  File "./li", line 34, in build_docker_image_if_missing
    ret = subprocess.run("docker image inspect lithium_docker_image".split(" "), stdout=open(os.devnull, 'wb')).returncode
  File "/Users/sugizo/miniconda3/lib/python3.6/subprocess.py", line 403, in run
    with Popen(*popenargs, **kwargs) as process:
  File "/Users/sugizo/miniconda3/lib/python3.6/subprocess.py", line 709, in __init__
    restore_signals, start_new_session)
  File "/Users/sugizo/miniconda3/lib/python3.6/subprocess.py", line 1344, in _execute_child
    raise child_exception_type(errno_num, err_msg, err_filename)
FileNotFoundError: [Errno 2] No such file or directory: 'docker': 'docker'
```

**prerequisites**
- /bin/python3
- docker

**question**
is it possible to run lithium without docker on mac ?

thanks

**Comment by matt-42** [comment]
I'm on it. thanks for the report.

**Reply by sugizo** [reply]
**another suggestion**
better please check the python3 path first with **'which'** command if it in linux or mac and use the return value from which command
because some users use python3 with conda or miniconda because they dont want to intact default python installed automatically by os
**e.g.**
```
$ which python3
/Users/sugizo/miniconda3/bin/python3
```

thanks

**Reply by matt-42** [reply]
I fixed it in my last commit.

**Reply by sugizo** [reply]
retested again still got the same result (docker must be installed on system)

**Reply by matt-42** [reply]
Sorry I misread you question. Yes you can install lithium without docker: 
Use this script instead:  https://raw.githubusercontent.com/matt-42/lithium/master/install.sh 
Or if you don't use the symbol generation [1] you just need to download the headers in https://github.com/matt-42/lithium/tree/master/single_headers (lithium is header only)

The libraries required by the hello world example are: 
   - ssl
   - crypto
   - boost_context

[1] https://matt-42.github.io/lithium/getting-started/automatic-symbols-generation
